### ISSUE TITLE
ldap tls protection "expires" after some time

### ISSUE THREAD

**Issue by juckerf** [gyselroth/kube-ldap#13]
after running kube-ldap for some time, the tls protection on the ldap connection "expires" (probably the whole connection times out), leading to the following error in the log (& the inability to authenticate):
`2018-05-17T07:27:53.666Z - info: ldap connection not tls protected`

**Comment by shackit** [comment]
Also experiencing a similar issue where after some time kube-ldap would stop authentication and ultimately stop providing the JWT token - after adding some additional logging I could observe error messages like below. After this no attempt was made to reconnect.

`2018-07-17T15:43:57.204Z - error: uncaughtException: read ECONNRESET date=Tue Jul 17 2018 15:43:57 GMT+0000 (UTC), pid=1, uid=0, gid=0, cwd=/, execPath=/usr/local/bin/node, version=v8.10.0, argv=[/usr/local/bin/node, /srv/www/kube-ldap/build/index.js], rss=43806720, heapTotal=16072704, heapUsed=13692136, external=25028, loadavg=[0, 0, 0], uptime=30241, trace=[column=11, file=util.js, function=_errnoException, line=1022, method=null, native=false, column=25, file=net.js, function=TCP.onread, line=628, method=onread, native=false], stack=[Error: read ECONNRESET,     at _errnoException (util.js:1022:11),     at TCP.onread (net.js:628:25)]`

This https://github.com/joyent/node-ldapjs/issues/392 shows a method to configure the ldap client to reconnect on failure.

Updating `app.js` with the following has helped kube-ldap to re-establish connectivity to the underlying LDAP directory.

```
ldap.createClient({
    url: config.ldap.uri,
    timeout: config.ldap.timeout * 1000,
    connectTimeout: config.ldap.timeout * 1000,
    log: ldapLog,
    reconnect: {
      initialDelay: 100,
      maxDelay: 1000,
      failAfter: 10
    }
  }),
```

**Comment by juckerf** [comment]
great work @shackit :+1: 
I did not find the time to debug this issue further, but I guess your solution should solve the issue.
I'll add the `reconnect` configuration to kube-ldap as soon as possible. or you can submit a PR if you want to speed up the integration of said configuration in kube-ldap :)

**Comment by juckerf** [comment]
@shackit thanks again! FYI the new docker image is available at the tag v1.2.1 (or latest)
